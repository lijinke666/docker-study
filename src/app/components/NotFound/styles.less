.space{
    margin:0 5px;
}