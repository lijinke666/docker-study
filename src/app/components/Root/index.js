import React, { PureComponent } from "react";

export default class Root extends PureComponent {
  constructor(props) {
    super(props);
  }
  render() {
    return <div>{this.props.children}</div>;
  }
}
