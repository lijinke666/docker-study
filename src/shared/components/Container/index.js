/*
 * @Author: jinke.li 
 * @Date: 2017-07-17 19:41:54 
 * @Last Modified by:   jinke.li 
 * @Last Modified time: 2017-07-17 19:41:54 
 */
import React from "react";
import classNames from "classnames";
import "./index.less";

//默认宽度 为1200px 所以将你的html 外面包一个这个 <Container>你的html</Container>

export default class Container extends React.PureComponent {
  render() {
    const { className, ...attr } = this.props;
    return (
      <div key="container" className="container">
        <div className={classNames("wrap", className)} {...attr}>
          {this.props.children}
        </div>
      </div>
    );
  }
}
