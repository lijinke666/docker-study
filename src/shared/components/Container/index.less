@import "~styles/vars.less";
.container{
    &::before,&::after{
        content:"";
        clear: both;
        zoom: 1;
    }
    .wrap{
        width:@main-width;
        @media screen and (max-width:768px){
            width:100vw;
        }
        margin:0 auto;
    }
}