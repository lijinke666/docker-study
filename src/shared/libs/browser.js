//项目中可能需要用到判断用户是手机还是pc

const browser = {
  isPC() {
    let userAgentInfo = navigator.userAgent;
    let Agents = [
      "Android",
      "iPhone",
      "SymbianOS",
      "Windows Phone",
      "iPad",
      "iPod"
    ];
    let flag = true;
    for (let v = 0; v < Agents.length; v++) {
      if (userAgentInfo.indexOf(Agents[v]) > 0) {
        flag = false;
        break;
      }
    }
    return flag;
  },
  language: (navigator.browserLanguage || navigator.language).toLowerCase()
};

export default browser;
