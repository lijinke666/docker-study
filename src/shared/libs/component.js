/**
 * 获取高阶组件名字
 * @returns name
 */
export function getDisplayName(component) {
  return component.displayName || component.name || "Component";
}
