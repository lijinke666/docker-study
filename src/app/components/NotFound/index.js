import React from "react";
import { Link } from "react-router-dom";
import { Alert } from "antd";
import history from "libs/history";
import "./styles.less";

const NotFound = () => (
  <Alert
    message="404"
    description={
      <>
        <p>页面去火星了,暂时失联了.</p>
        <p>
          <a onClick={() => history.go(-1)}>回到上一级</a>
          <span className="space" />
          <Link to="/">回到首页</Link>
        </p>
      </>
    }
    type="info"
    showIcon
  />
);

export default NotFound;
