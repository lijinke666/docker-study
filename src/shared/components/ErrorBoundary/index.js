import React, { Component } from "react";
import { Alert } from "antd";
import { getDisplayName } from "libs/component";

/**
 * @name ErrorBoundary
 * @use  @ErrorBoundary
 */

export default function ErrorBoundary(WrappedComponent) {
  return class HOC extends Component {
    static displayName = `HOC(${getDisplayName(WrappedComponent)})`;
    constructor(props) {
      super(props);
      this.state = { error: null, errorInfo: null };
    }

    componentDidCatch(error, errorInfo) {
      this.setState({
        error,
        errorInfo
      });
    }
    render() {
      if (this.state.errorInfo) {
        return (
          <Alert
            message="ERROR:"
            description={
              <div>
                {this.state.error && this.state.error.toString()}
                <br />
                {this.state.errorInfo.componentStack}
              </div>
            }
            type="error"
            showIcon
          />
        );
      }
      return <WrappedComponent {...this.props} />;
    }
  };
}
