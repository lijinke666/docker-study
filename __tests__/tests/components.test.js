/*eslint-disable */
import React from "react";
import assert from "power-assert";
import Root from "../../src/app/components/Root";
import ErrorBoundary from "../../src/shared/components/ErrorBoundary";
import Container from "../../src/shared/components/Container";
import NotFound from "../../src/app/components/NotFound";
import {getDisplayName} from "../../src/shared/libs/component";
import { Alert } from "antd";
import { render, mount, shallow } from "enzyme";

describe("ComponentsTests", () => {
  describe("<Root/>", () => {
    it("should render <Root/>", () => {
      const wrapper = shallow(<Root />);
      expect(wrapper).toMatchSnapshot();
      assert(wrapper.find("div").length >= 1);
    });
  });
  describe("<NotFound/>", () => {
    it("should render <NotFound/>", () => {
      const wrapper = shallow(<NotFound />);
      expect(wrapper).toMatchSnapshot();
      assert(wrapper.find(Alert).length === 1);
    });
  });
  describe("<ErrorBoundary/>", () => {
    it("should render <Alert/> of the <ErrorBoundary/>", () => {
      const MyComponent = ErrorBoundary(Root);
      const wrapper = mount(<MyComponent />);
      wrapper.setState({ errorInfo: "errorInfo", error: "error" });
      assert(wrapper.find(Alert).length === 1);
    });
  });
  describe("<Container/>", () => {
    it("should render children of the <Container/> and have container className", () => {
      const wrapper = shallow((
        <Container>
          <div className="dawdler" />
        </Container>
      ));
      assert( wrapper.contains(<div className="dawdler" />) === true)
      assert( wrapper.find(".container").length === 1)
      assert( wrapper.find(".wrap").length === 1)
    });
  });
  describe('getDisplayName',()=>{
    it('should return components getDisplayName',()=>{
      const name = getDisplayName(Container)
      assert( name === "Container")
    })
  })
});
