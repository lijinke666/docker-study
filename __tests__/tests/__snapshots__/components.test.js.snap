// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentsTests <NotFound/> should render <NotFound/> 1`] = `
<Alert
  description={
    <UNDEFINED>
      <p>
        页面去火星了,暂时失联了.
      </p>
      <p>
        <a
          onClick={[Function]}
        >
          回到上一级
        </a>
        <span
          className="space"
        />
        <Link
          replace={false}
          to="/"
        >
          回到首页
        </Link>
      </p>
    </UNDEFINED>
  }
  message="404"
  showIcon={true}
  type="info"
/>
`;

exports[`ComponentsTests <Root/> should render <Root/> 1`] = `<div />`;
